gladier_kanzus/tools/gather_data.py

```diff
@@ -1,97 +1,101 @@
-        # match int-0-Kaleidoscope_15_05189.pickle
-        int_match = re.match(r'int-(\d+)-(\w+)_(\d+)_(\d+).pickle', filename)
+
+    for int_file in glob.glob(os.path.join(proc_dir,'int-*.pickle')):
+        int_file = os.path.basename(int_file)
+        int_match = re.match(r'int-\d+-\w+_\d+_(\d+).pickle', int_file)
         if int_match:
-            int_filenames.append(filename)
-            int_number, exp_name, beamline_run, int_index = int_match.groups()
+            int_index = int_match.groups()[0]
+            int_filenames.append(int_file)
             int_indices.append(int(int_index))
-        # else:
-        #     match idx-Kaleidoscope_15_00001_datablock.json
-        #     cbf_match = re.match(r'idx-\w+_\d+_(\d+)_datablock.json', filename)
-        #     cbf_match = re.match(r'idx-\w+_\d+_(\d+)_datablock.json', filename)
-        #     if cbf_match:
-        #         cbf_index = int(cbf_match.groups()[0])
-        #         cbf_indices.append(cbf_index)
-        if '.log' in filename:
-            with open(filename,'r') as f:
-                for line in f.readlines():
-                    match = re.findall(r"(\d+).cbf", line)
-                    if match:
-                        cbf_index = int(match[0])
-                        cbf_indices.append(cbf_index)
+
+    for log_file in glob.glob(os.path.join(proc_dir,'log*.txt')):
+        with open(log_file,'r') as f:
+            for line in f.readlines():
+                match = re.findall(r"\w+_\d+_(\d+).cbf", line)
+                if match:
+                    cbf_index = int(match[0])
+                    cbf_indices.append(cbf_index)
     
     if len(cbf_indices) == 0:
         cbf_indices.append(0)
+   
+    proc_cbf_file = os.path.join(proc_dir,'proc_cbf.txt')
+    
+    if os.path.exists(proc_cbf_file):
+        os.remove(proc_cbf_file)
+
+    with open(proc_cbf_file,'w+') as f:
+        for cbf in sorted(cbf_indices):
+            f.write(str(cbf) + "\n")
+
+    proc_ints_file = os.path.join(proc_dir,'proc_ints.txt')
+    
+    if os.path.exists(proc_ints_file):
+        os.remove(proc_ints_file)
+
+    with open(proc_ints_file,'w+') as f:
+        for intfile in sorted(int_filenames):
+            f.write(str(intfile) + "\n")
     
     batch_info = {
         'cbf_files': len(cbf_indices),
         'cbf_file_range': {'from': min(cbf_indices), 'to': max(cbf_indices)},
         'total_number_of_int_files': len(int_indices)
     }
 
-    # Move int files to the upload dir.
-    os.makedirs(tar_input, mode=0o775, exist_ok=True)
-    for int_filename in int_filenames:
-        shutil.copyfile(os.path.join(processing_dir, int_filename),
-                        os.path.join(tar_input, int_filename))
-
     # Fetch beamline metadata
     with open(beamline_file, 'r') as fp:
         beamline_metadata = json.load(fp)
+
     user_input = beamline_metadata.get('user_input', {})
     protein = user_input.get('protein_name', protein)
 
+    metadata = {
```

Comment:
@NickolausDS any objections into moving this here and if necessary putting new info on `**data` on the client level?

Reply:
Two would be readability and configurability. For readability, it's nice to have as a top level input item so newer folks know where stuff is going, and don't need to dig into the code to find out. For configurability, it would be good if we ever want to have a 'test' flow that uses different values for `index`, `dataset`, `source_globus_endpoint`, etc.

Reply:
I know it kinda 'pollutes' the input with values that rarely ever change, but it would be harder to change them if we needed to, especially for new users not familiar with the codebase.

Reply:
I will move then to the input top-level. I mostly tried to avoid nested dict's on the main payload.

Reply:
It does make the input uglier yeah.